ldms/python/ldmsd/parser_util.py

```diff
@@ -169,27 +169,35 @@ def parse_yaml_bool(bool_):
         return False
 
 def perm_handler(perm_str):
-    if perm_str is None or type(perm_str) is int:
+    if perm_str is None:
         return perm_str
-    nperm = "0"
-    perm_str = perm_str.split('-')
-    if len(perm_str) == 1:
+    if type(perm_str) is not str:
+        raise TypeError(f'Error: YAML "perms" value must be a string')
+    nperm = '0'
+    if len(perm_str.split('-')) > 1:
+        if len(perm_str) != 9:
+            raise ValueError(f'Error: Parsing YAML permisson string "{perm_str}"\n'
+                             f'When using unix-like vernacular to configure permissions you must account for read/write in user/group/global\n'
+                             f'Allowed format: (r|-)(w|-)-(r|-)(w|-)-(r|-)(w|-)-')
+        i = 0
+        x = 0
+        for ch in perm_str:
```

Comment:
And there isn't checking to see if the letters are correctly "r" or "w, or that the "x" positions are "-". So this would be accepted right now

```Qf-ltoxep```

No bits set, of course, but I think that needs to raise an error.:)

I think I would probably just use a regex something like below. The regex takes care of all the correctness checking, and we just need to sum up the bit values.

```
perms_pattern = re.compile('(?:(r)|-)(?:(w)|-)-(?:(r)|-)(?:(w)|-)-(?:(r)|-)(?:(w)|-)-')
m = perms_pattern.match(sys.argv[1])
if not m:
    raise ValueError("whatever")
perms = 0
if m.group(1):
    perms += 0o400
if m.group(2):
    perms += 0o200
if m.group(3):
    perms += 0o040
if m.group(4):
    perms += 0o020
if m.group(5):
    perms += 0o004
if m.group(6):
    perms += 0o002
perm_str = '0'+oct(perms)[2:]
```